src/transactions/transactions.util.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  BadRequestException,
+  InternalServerErrorException,
+  NotFoundException,
+  UnprocessableEntityException,
+} from '@nestjs/common';
+import {
+  ErrorCode,
+  GenericPolymeshTransaction,
+  NoArgsProcedureMethod,
+  ProcedureMethod,
+  ProcedureOpts,
+} from '@polymeshassociation/polymesh-sdk/types';
+import {
+  isPolymeshError,
+  isPolymeshTransaction,
+  isPolymeshTransactionBatch,
+} from '@polymeshassociation/polymesh-sdk/utils';
+
+import { BatchTransactionModel } from '~/common/models/batch-transaction.model';
+import { TransactionQueueModel } from '~/common/models/transaction-queue.model';
+import { TransactionModel } from '~/common/models/transaction.model';
+import { ModelResolver } from '~/common/utils';
+
+export type QueueResult<T> = {
+  result: T;
+  transactions: (TransactionModel | BatchTransactionModel)[];
+};
+
+export type WithArgsProcedureMethod<T> = T extends NoArgsProcedureMethod<unknown, unknown>
+  ? never
+  : T;
+
+/**
+ * a helper function to handle when procedures have args and those without args
+ */
+export function prepareProcedure<MethodArgs, ReturnType, TransformedReturnType = ReturnType>(
```

Comment:
This is messy and untype safe, and is the main reason I put this up as a draft for now.

We did have some TODO comments about handling no args procedure throughout the code. I'll try to have another look at this to see if I can come up with something better.

The challenge is the overloaded signature where opts need to be passed in first when interacting with a NoArgs procedure

Reply:
The types are better now, albeit a bit verbose.